### DISCUSSION TITLE
Why not use a rust library for v4.local (or StableLib XChacha20 library)?

### DISCUSSION THREAD

**Discussion by Extarys** [panva/paseto#41] [Q&A]
Out of curiosity, mostly, but also because symmetric encryption tend to be faster/more efficient. This could be a simple command to compile the library on the user's machine, if the user wish to use vX.local that isn't currently supported by only using JS.

EDIT I also found a pure TS implementation of XChacha20here https://github.com/StableLib/stablelib/tree/master/packages/xchacha20

**Comment by panva** [comment]
https://github.com/panva/paseto/discussions/54#discussioncomment-8698739
